Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Ride.hs

```diff
@@ -161,3 +164,90 @@ cancelRides rideIds now = do
         RideUpdatedAt =. val now
       ]
     where_ $ tbl ^. RideTId `in_` valList (toKey <$> rideIds)
+
+data RideItem = RideItem
+  { rideShortId :: ShortId Ride,
+    rideCreatedAt :: UTCTime,
+    rideId :: Id Ride, 
+    customerName :: Maybe Text, 
+    person :: Person, 
+    ride :: Ride,
+    bookingStatus :: Common.BookingStatus
+  }
+
+findAllRideItems ::
+  Transactionable m =>
+  Id Merchant ->
+  Int ->
+  Int ->
+  Maybe Common.BookingStatus ->
+  Maybe (ShortId Ride) -> 
+  Maybe DbHash ->
+  Maybe Text ->
+  UTCTime ->
+  m [RideItem]
+findAllRideItems merchantId limitVal offsetVal mbBookingStatus mbRideShortId mbCustomerPhoneDBHash mbDriverPhone now = do
+  res <- Esq.findAll $ do
+    booking :& ride :& person <-
+      from $
+        table @BookingT
+          `innerJoin` table @RideT
+            `Esq.on` ( \(booking :& ride) ->
+                         ride ^. Ride.RideBookingId ==. booking ^. Booking.BookingTId
+                     )
+          `innerJoin` table @PersonT
+            `Esq.on` ( \(booking :& _ :& person) ->
+                         booking ^. Booking.BookingRiderId ==. person ^. Person.PersonTId
+                     )
+    let bookingStatusVal = mkBookingStatusVal ride
+    where_ $
+      booking ^. BookingMerchantId ==. val (toKey merchantId)
+        &&. whenJust_ mbBookingStatus (\bookingStatus -> bookingStatusVal ==. val bookingStatus)
+        &&. whenJust_ mbRideShortId (\rideShortId -> ride ^. Ride.RideShortId ==. val rideShortId.getShortId)
+        &&. whenJust_ mbCustomerPhoneDBHash (\hash -> person ^. Person.PersonMobileNumberHash ==. val (Just hash))
+        &&. whenJust_ mbDriverPhone (\driverMobileNumber -> ride ^. Ride.RideDriverMobileNumber ==. val driverMobileNumber)
+    orderBy [desc $ ride ^. RideCreatedAt]
+    limit $ fromIntegral limitVal
+    offset $ fromIntegral offsetVal
+    return
+      ( ride ^. RideShortId,
+        ride ^. RideCreatedAt,
+        ride ^. RideTId,
```

Comment:
you don't need above three fields if your already have `ride` entity  on line 218